quartz_solar_forecast/forecast.py

```diff
@@ -30,6 +26,110 @@ def run_forecast(site: PVSite, ts: datetime | str = None, nwp_source: str = "ico
     pv_xr = make_pv_data(site=site, ts=ts)
 
     # load and run models
-    pred_df = forecast_v1(nwp_source, nwp_xr, pv_xr, ts)
+    pred_df = forecast_v1(nwp_source, nwp_xr, pv_xr, ts, model=model)
 
     return pred_df
+
+
+def predict_tryolabs(
+    site: PVSite, model=None, ts: datetime | str = None):
+    """Run the forecast with the tryolabs model"""
+    solar_power_predictor = TryolabsSolarPowerPredictor(model=model)
+
+    if ts is None:
+        start_date = pd.Timestamp.now().strftime("%Y-%m-%d")
+        start_time = pd.Timestamp.now().floor("15min")
+    else:
+        start_date = pd.Timestamp(ts).strftime("%Y-%m-%d")
+        start_time = pd.Timestamp(ts).floor("15min")
+  
+    end_time = start_time + pd.Timedelta(hours=48)
+
+    predictions = solar_power_predictor.predict_power_output(
+        latitude=site.latitude,
+        longitude=site.longitude,
+        start_date=start_date,
+        kwp=site.capacity_kwp,
+        orientation=site.orientation,
+        tilt=site.tilt,
+    )
+
+    if predictions is not None:
+        predictions = predictions[
+            (predictions["date"] >= start_time) & (predictions["date"] < end_time)
+        ]
+        predictions = predictions.reset_index(drop=True)
+        predictions.set_index("date", inplace=True)
+        print("Predictions finished.")
+        return predictions
+
+
+def download_model(filename: str, file_id: str):
+    """
+    Download model from google drive.
+
+    Parameters
+    ----------
+    filename : str
+        The name of the model to be saved
+    file_id: 
+        Google id of the model file
+    """
+    gdown.download(f'https://drive.google.com/uc?id={file_id}', filename, quiet=False)
+
+
+def decompress_zipfile(filename: str):
+    """
+    Extract all files contained in a .zip file to the current directory.
+    filename must contain .zip extension
+
+    Parameters
+    ----------
+    filename : str
+        The name of the .zip file to be decompressed
+    """
+    with zipfile.ZipFile(filename, "r") as zip_file:
+        zip_file.extractall()
+
+
+def run_forecast(
+    site: PVSite,
+    model: str = "tryolabs",
```

Comment:
could we leave it as ocf by default for the moment? and we can later think about changing it over

Reply:
Changed default model to "ocf"